### DISCUSSION TITLE
Any plans to support XCFrameworks?

### DISCUSSION THREAD

**Discussion by dvkellerman** [swiftyfinch/Rugby#35] [Ideas]
It would be good to have binary frameworks for x86 and arm64 archs in one file.

**Comment by swiftyfinch** [comment]
Hey @dvkellerman, thanks for open this discussion!

The main goal of 🏈 Rugby is optimization.
If we build both SDKs, it will take 2x longer. But you need only one at the present time.
You can switch between SDKs faster enough and all cache will be saved for each SDKs appropriately.

Do you think it's anyway important feature for you?

**Reply by muukii** [reply]
What purpose is creating XCFramework?
If we want to get caching, no need to create XCFramework, need the building cache for each arch instead.

Creating and using a prebuilt framework make the Pods project confusing. 
the current approach is beautiful (rewriting xcconfig file for each configuration).

**Reply by swiftyfinch** [reply]
Yeah, I think you are right. XCFramework is not a solution for building both ARCHs with Rugby, but it is just a convenient way to merge it into one entity. It can be useful in a different way.

**Comment by swiftyfinch** [comment]
🚛 I added [the task](https://github.com/swiftyfinch/Rugby/projects/3#card-65777632) for this feature request

**Reply by shahzadmajeed** [reply]
Hi, sorry for bringing this thread back to life but was this expected:

"Creating cache for one sdk (ios for example) resets existing cache for other sdk (sim in this case)" 
This results into build errors which means every time you are switching b/w device & sim you are going to build pods for current sdk/arch. 

Isn't that frustrating? Could there be a solution?

I think XCFramework suggestion above isn't too bad after all. You can use a remote git repository to distribute these XCFrameworks once (built by CI/CD or one of your team members) and all team members can use same binaries. Just like https://github.com/grab/cocoapods-binary-cache

**Reply by swiftyfinch** [reply]
Hi! Yeah, you are right. We just need time to add this feature as an alternative way of using 🏈 Rugby.
Sorry, but I haven't it for now. Maybe I will find time at the end of this year.

**Comment by swiftyfinch** [comment]
#58 Done

**Reply by shahzadmajeed** [reply]
That is amazing. Thank you for adding support to build both device & sim at the same time. I think it will boost productivity of plugin users as I see us building main app targets more often than Pod dependencies.
